### PR TITLE
Import export

### LINKED ISSUE #15: Add possibilities to export and import of drawings

### PR DESCRIPTION
**Pull request by 0x4d**
Closes #15 #51 

## Describe your changes
Import and export modules have been reworked to use new module structure.

### CODE REVIEW

File: src/Viewer.js [hunk 1]
```diff
@@ -55,9 +55,17 @@ export default class Viewer extends Diagram {
...
+    }
+
+    on (eventName, callback) {
+        this.get('eventBus').on(eventName, callback);
+    }
+
     addFormalism (plugin) {
         const pluginInstance = this.injector.instantiate(plugin);
-        this.get('eventBus').fire('plugin.register', {
+        this.fire('plugin.register', {
```
Comment (tim-kilian) [inline src/Viewer.js:72]: I like this.

File: src/core/ElementFactory.js [hunk 2]
```diff
@@ -9,10 +9,10 @@ export class ElementFactory extends BaseElementFactory {
     }
 
     create (type, attrs) {
-        return super.create(type, Object.assign(attrs, {
+        return super.create(type, Object.assign({
             id: uid(type),
             type: type,
-        }));
+        }, attrs));
```
Comment (tim-kilian) [inline src/core/ElementFactory.js:15]: Thanks for this fix, I didn't think about customizing the id / type.

**Review by tim-kilian — APPROVED** [review]
Good to see the import/export in the new module style and covered with tests. Thanks.

I am not that happy about the lodash dependency, but it's okay for me. I'll take it, when it satisfy your needs.

Over all its a great improvement. (y)

**Comment by 0x4d** [comment]
> I am not that happy about the lodash dependency, but it's okay for me. I'll take it, when it satisfy your needs.

Even though `JSON.parse(JSON.stringify())` hasn't caused any problems in our code until now, it could do so in the future by dropping any property that doesn't have a corresponding JSON type (like functions, undefined, NaN, Infinity, etc.). Abusing JSON to deep clone objects just feels hacky to me.

In the past, I always used lodash for this purpose as it is modular, tested and well established. However, I am of course open to better suggestions.

**Pull request merged**